test/lifecycle_test.dart

```diff
@@ -354,6 +354,80 @@ main() {
             reason: 'applies the state returned by `getDerivedStateFromError`');
       });
 
+      test('when error lifecycle getDerivedStateFromError returns null state should not update', () {
+        final Map initialState = {
+          'originalState': true,
+        };
+        var _shouldThrow = true;
+        final Map initialProps = unmodifiableMap({
+          'initialState': (_) => initialState,
+          'getDerivedStateFromError': (_, __) => null,
+          'render': (_) {
+            if (_shouldThrow) {
+              _shouldThrow = false;
+              return components2.ErrorComponent({"key": "errorComp"});
+            }
+            return react.div({});
+          }
+        });
+
+        LifecycleTestHelper component = getDartComponent(render(components2.ErrorLifecycleTest(initialProps)));
+
+        expect(
+            component.lifecycleCallMemberNames,
+            equals([
+              'defaultProps',
+              'initialState',
+              'getDerivedStateFromProps',
+              'render',
+              'getDerivedStateFromError',
+              'render',
+              'componentDidMount',
+              'componentDidCatch'
+            ]));
+        expect(component.state, initialState,
+            reason: 'when `getDerivedStateFromError` returns null it should not update state.');
```

Comment:
#nit this seems redundant with the test description - although I think it reads better than the test description itself 😝